src/binary/uint.rs

```diff
@@ -18,20 +17,30 @@ impl DecodedUInt {
     /// If the length of `uint_bytes` is greater than the size of a `u128`, returns `Err`.
     #[inline]
     pub(crate) fn uint_from_slice(uint_bytes: &[u8]) -> IonResult<u128> {
-        if uint_bytes.len() > mem::size_of::<u128>() {
-            return IonResult::decoding_error(
+        const MAX_BYTES: usize = size_of::<u128>();
+        // The `uint_from_slice_unchecked` method will work for a uint of any size up to and including
+        // 16 bytes. However, because the slice is of an unknown length, the `memcpy` performed by
+        // that method does not get inlined by the compiler. Here we check for some common int sizes
+        // and construct the int manually to allow the hot path to be inlined.
+        match uint_bytes.len() {
```

Comment:
No case needed for `0`?

Reply:
I'd assume that's handled in a different layer and would skip calling this method entirely, but if so that would be a good thing to comment at least.

Reply:
> No case needed for 0?

As it was written, the empty slice would use the general-purpose path. I didn't add a special case for it because I was concerned about the code for `uint_from_slice` itself getting to big. However, I just ran my benchmarks with the special case for `0` added and it doesn't seem to have pushed it over that size threshold, so let's do it!